sphinx/ext/napoleon/docstring.py

```diff
@@ -54,6 +53,32 @@
 _SINGLETONS = ("None", "True", "False", "Ellipsis")
 
 
+class _Stack:
```

Comment:
I'm not sure the "Stack" is the best name for this purpose. AFAIK, it's usually used for the LIFO collection. But the lines are not so.

Reply:
It's LIFO if you see peek()ing as taking the last element and then putting it back.  I don't have a strong opinion about the name; the main point is that the data structure needed here is basically just a list (using python's ability to easily pop from the list's end).  Alternatively, I could have used a collections.deque to implement the data structure (keeping the lines in logical order and using popleft()), but that doesn't buy us much, requires using a less well-known class, and direct indexing would be a bit less efficient.

Reply:
I'd vote for `collections.deque`.

A